### PR TITLE
VZ-6572: Update base profiles to use v1beta1 API

### PR DESCRIPTION
**Pull request by Abhijithang**
This PR will update the base profiles to use v1beta1 API and refactors the profiles to their respective API directories.

### CODE REVIEW

File: platform-operator/internal/config/config.go [hunk 1]
```diff
@@ -172,8 +172,14 @@ func GetProfilesDir() string {
 	return filepath.Join(instance.VerrazzanoRootDir, profilesDirSuffix)
 }
 
-func GetProfile(profile string) string {
-	return filepath.Join(GetProfilesDir(), profile+".yaml")
+// GetProfileV1alpha1 returns v1alpha1 API profiles dir
+func GetProfileV1alpha1(profile string) string {
```
Comment (anders-swanson) [inline platform-operator/internal/config/config.go:176]: thought, can you write a function that is like
```
func GetProfile(groupVersion schema.GroupVersion, profile string) {
...
}
```
And the profile path is known from the GroupVersion information? Every API has a GroupVersion object exported.

Comment (Abhijithang) [reply platform-operator/internal/config/config.go:176]: Makes sense, thanks Anders. Will do that now.

Comment (Abhijithang) [reply platform-operator/internal/config/config.go]: Done.

File: platform-operator/manifests/profiles/v1beta1/base.yaml [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+                      topologyKey: kubernetes.io/hostname
+                    weight: 100
+    dns:
+      wildcard:
+        domain: "nip.io"
+    elasticsearch:
+      enabled: true
+    fluentd:
+      enabled: true
+      elasticsearchURL: "http://verrazzano-authproxy-elasticsearch:8775"
```
Comment (anders-swanson) [inline platform-operator/manifests/profiles/v1beta1/base.yaml:87]: This is not v1beta1

File: platform-operator/manifests/profiles/v1beta1/base.yaml [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+                    affinity:
+                      podAntiAffinity:
+                        preferredDuringSchedulingIgnoredDuringExecution:
+                          - podAffinityTerm:
+                              labelSelector:
+                                matchLabels:
+                                  app: istiod
+                              topologyKey: kubernetes.io/hostname
+                            weight: 100
+      ingress:
```
Comment (anders-swanson) [inline platform-operator/manifests/profiles/v1beta1/base.yaml:141]: this is not v1beta1

**Review by anders-swanson — CHANGES_REQUESTED** [review]
Can you evaluate the v1beta1 profiles again with the changes that have been made in the API, and write some tests which validate they are as expected?

File: platform-operator/manifests/profiles/v1beta1/base.yaml (hunk 2 above)
Comment (Abhijithang) [reply platform-operator/manifests/profiles/v1beta1/base.yaml]: Made the changes.

File: platform-operator/manifests/profiles/v1beta1/base.yaml (hunk 3 above)
Comment (Abhijithang) [reply platform-operator/manifests/profiles/v1beta1/base.yaml]: > Can you evaluate the v1beta1 profiles again with the changes that have been made in the API, and write some tests which validate they are as expected?

Sure, evaluated and added the unit tests. Thanks.

File: platform-operator/apis/verrazzano/v1alpha1/testdata/base/v1alpha1.yaml [hunk 4]
```diff
@@ -0,0 +1,235 @@
+# Copyright (c) 2021, 2022, Oracle and/or its affiliates.
```
Comment (anders-swanson) [inline platform-operator/apis/verrazzano/v1alpha1/testdata/base/v1alpha1.yaml:1]: `Copyright (c) 2022,`

File: platform-operator/apis/verrazzano/v1alpha1/testdata/base/v1beta1.yaml [hunk 5]
```diff
@@ -0,0 +1,256 @@
+# Copyright (c) 2021, 2022, Oracle and/or its affiliates.
```
Comment (anders-swanson) [inline platform-operator/apis/verrazzano/v1alpha1/testdata/base/v1beta1.yaml:1]: fix copyright

File: platform-operator/manifests/profiles/v1beta1/prod.yaml [hunk 6]
```diff
@@ -0,0 +1,38 @@
+# Copyright (c) 2022, Oracle and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+spec:
+  components:
+    opensearch:
+      enabled: true
+      installArgs:
```
Comment (anders-swanson) [inline platform-operator/manifests/profiles/v1beta1/prod.yaml:8]: these have to be OpenSearch Nodes. I meant not to include the default _empty_ node array created by conversion.

File: platform-operator/manifests/profiles/v1beta1/dev.yaml [hunk 7]
```diff
@@ -0,0 +1,16 @@
+# Copyright (c) 2022, Oracle and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+spec:
+  defaultVolumeSource:
+    emptyDir: { }
+  components:
+    opensearch:
+      installArgs:
```
Comment (anders-swanson) [inline platform-operator/manifests/profiles/v1beta1/dev.yaml:8]: Need to include nodes (can skip empty nodes)

File: platform-operator/apis/verrazzano/v1alpha1/testdata/prod/v1beta1.yaml [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+          replicas: 3
+          resources:
+            requests:
+              memory: 1.4Gi
+          roles:
+            - master
+          storage:
+            size: 50Gi
+      enabled: true
+      installArgs:
```
Comment (anders-swanson) [inline platform-operator/apis/verrazzano/v1alpha1/testdata/prod/v1beta1.yaml:37]: there should be no install args in v1beta1

File: platform-operator/apis/verrazzano/v1alpha1/testdata/dev/v1beta1.yaml [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+          roles:
+          - ingest
+        - name: es-master
+          replicas: 1
+          resources:
+            requests:
+              memory: 1G
+          roles:
+          - master
+      installArgs:
```
Comment (anders-swanson) [inline platform-operator/apis/verrazzano/v1alpha1/testdata/dev/v1beta1.yaml:26]: no install args in v1beta1

File: platform-operator/apis/verrazzano/v1alpha1/testdata/base/v1beta1.yaml (hunk 5 above)
Comment (Abhijithang) [reply platform-operator/apis/verrazzano/v1alpha1/testdata/base/v1beta1.yaml]: Done.

File: platform-operator/manifests/profiles/v1beta1/prod.yaml (hunk 6 above)
Comment (Abhijithang) [reply platform-operator/manifests/profiles/v1beta1/prod.yaml]: Include the nodes, need you to verify once again.

File: platform-operator/manifests/profiles/v1beta1/dev.yaml (hunk 7 above)
Comment (Abhijithang) [reply platform-operator/manifests/profiles/v1beta1/dev.yaml]: Done. Need your review again.

File: platform-operator/apis/verrazzano/v1alpha1/testdata/prod/v1beta1.yaml (hunk 8 above)
Comment (Abhijithang) [reply platform-operator/apis/verrazzano/v1alpha1/testdata/prod/v1beta1.yaml]: Removed.

File: platform-operator/apis/verrazzano/v1alpha1/testdata/dev/v1beta1.yaml (hunk 9 above)
Comment (Abhijithang) [reply platform-operator/apis/verrazzano/v1alpha1/testdata/dev/v1beta1.yaml]: Sorry about that, done.

**Review by anders-swanson — APPROVED** [review]

**Review by anders-swanson — APPROVED** [review]

**Pull request merged**